contracts/PrimeCasino.sol

```diff
@@ -1,19 +1,167 @@
+  	(endTime ,,) = enforcer.getStatus(taskHash);
+  	require(endTime > 0, "not a known bet");
+  	require(endTime == 1 || endTime > now, "bet duration expired");
+    Sum memory sum = primeBetSums[_candidateNumber];
+  	if (_isPrime) {
+  		primeBets[_candidateNumber][msg.sender] += int256(msg.value);
+  		sum.sumYes += msg.value;
+  	} else {
+  		primeBets[_candidateNumber][msg.sender] -= int256(msg.value);
+  		sum.sumNo += msg.value;
+  	}
+    emit NewBet(_candidateNumber, taskHash, sum.sumYes, sum.sumNo);
+  }
+
+  /**
+   * @dev Returns the status for a given `candidateNumber` that is a potentially prime.
+   *
+   * the return values should be interpreted like this:
+   * 
+   * |                       | challengeEndTime > now                                             | challengeEndTime <= now && challengeEndTime > 0                       | challengeEndTime == 0 |
+   * |-----------------------|--------------------------------------------------------------------|-----------------------------------------------------------------------|-----------------------|
+   * | pathRoots.length == 0 | no computation result registered yet. initial state after request. | computation market could not deliver any solution to given task.      | not requested         |
+   * | pathRoots.length == 1 | first result registered. open for challenges.                      | exactly 1 result found. result can be used.                           | not requested         |
+   * | pathRoots.length > 1  | some existing result challenged. cDance in progress.               | more than one result survived the cDance, computation market failed.  | not requested         |
+   *
+   */
+  function getStatus(uint256 _candidateNumber) public view returns (uint256 _challengeEndTime, bytes32[] memory _pathRoots) {
+    Sum memory sum = primeBetSums[_candidateNumber];
+    if (sum.sumYes == 0 && sum.sumNo == 0) {
+      // this number has not been requested yet
+      return (_challengeEndTime, _pathRoots);
+    }
+  	bytes32 taskHash = sum.taskHash;
+  	IEnforcer enforcer = IEnforcer(enforcerAddr);
+  	(_challengeEndTime, _pathRoots,) = enforcer.getStatus(taskHash);
+  }
 
+  /**
+   * @dev pays out `msg.sender` on `canditateNumber`.
+   *
+   * Requirements:
+   *
+   * - `msg.sender` should have bet on this number before.
+   * - `canditateNumber` should have passed challengeEndTime.
+   */
+  function payout(uint256 _candidateNumber) public {
+  	int256 betAmount = primeBets[_candidateNumber][msg.sender];
+  	require(betAmount != 0, "no bet amount");
+  	uint256 endTime;
+  	bytes32[] memory pathRoots;
+  	bytes32[] memory results;
+  	bytes32 taskHash = primeBetSums[_candidateNumber].taskHash;
+  	IEnforcer enforcer = IEnforcer(enforcerAddr);
+  	(endTime, pathRoots, results) = enforcer.getStatus(taskHash);
+  	require(endTime < now, "bet duration not passed");
+  	uint256 payoutAmount;
+  	if (pathRoots.length == 0 || pathRoots.length > 1) {
```

Comment:
I’ve bet on `yes` and `no` — total bet === 0, so i can’t get my money back if no result was proposed

Reply:
don't know what to do in that case 🤷‍♂